test/ussd_optout_rapidpro.test.js

```diff
@@ -8,25 +10,748 @@ describe("ussd_optout_rapidpro app", function() {
+                    state: "state_info_deleted",
+                    reply: [
+                        "All your info will be permanently deleted in the next 7 days. " +
+                        "We'll stop sending messages. Please select Next to continue:",
+                        "1. Next"
+                    ].join("\n")
+                })
+                .run();
+        });
+    });
+    describe("state_optout_reason", function() {
+        it("should display the list of options", function() {
+            return tester
+                .setup.user.state("state_optout_reason")
+                .input({session_event: "continue"})
+                .check.interaction({
+                    state: "state_optout_reason",
+                    reply: [
+                        "We'll stop sending msgs. Why do you want to stop your MC msgs?",
                         "1. Miscarriage",
                         "2. Baby was stillborn",
-                        "3. Baby died",
-                        "4. Messages not useful",
-                        "5. Other"
+                        "3. Baby passed away",
+                        "4. Msgs aren't helpful",
+                        "5. Other",
+                        "6. More"
                     ].join("\n")
                 })
                 .run();
         });
+        it("should display the other options when the more option is chosen", function() {
+            return tester
+                .setup.user.state("state_optout_reason")
+                .input("6")
+                .check.interaction({
+                    state: "state_optout_reason",
+                    reply: [
+                        "We'll stop sending msgs. Why do you want to stop your MC msgs?",
+                        "1. I prefer not to say",
+                        "2. Back"
+                    ].join("\n")
+                })
+                .run();
+        });
+        it("should go to state_loss_optout if any of the first 3 options are chosen", function() {
+            return tester
+                .setup.user.state("state_optout_reason")
+                .input("2")
+                .check.user.state("state_loss_optout")
+                .run();
+        });
+        it("should go to state_delete_research_info if any of the last 3 options are chosen", function() {
+            return tester
+                .setup.user.state("state_optout_reason")
+                .input("5")
+                .check.user.state("state_delete_research_info")
+                .run();
+        });
+        it("should display an error if an incorrect input is sent", function() {
+            return tester
+                .setup.user.state("state_optout_reason")
+                .input("foo")
+                .check.interaction({
+                    state: "state_optout_reason",
+                    reply: [
+                        "We'll stop sending msgs. Why do you want to stop your MC msgs?",
```

Comment:
This error doesn't seem to be any different to the question. If this is intentional, maybe we should change the test description to `should display the question if an incorrect input is sent` to make that clear.

Reply:
yes, that was the intention